src/actions.rs

```diff
@@ -283,41 +277,25 @@ impl EGraph {
 
                     let value = if let Some(out) = function.nodes.get(values) {
                         out.value
-                    } else if *make_defaults {
+                    } else {
                         let ts = self.timestamp;
-                        let out = &function.schema.output;
-                        match function.decl.default.as_ref() {
-                            None if out.name() == UnitSort.name() => {
-                                function.insert(values, Value::unit(), ts);
-                                Value::unit()
-                            }
-                            None if out.is_eq_sort() => {
-                                let value = Value {
-                                    #[cfg(debug_assertions)]
-                                    tag: out.name(),
-                                    bits: self.unionfind.make_set(),
-                                };
-                                function.insert(values, value, ts);
-                                value
-                            }
-                            Some(default) => {
-                                let default = default.clone();
-                                let value = self.eval_resolved_expr(&default)?;
-                                self.functions.get_mut(f).unwrap().insert(values, value, ts);
-                                value
-                            }
-                            _ => {
-                                return Err(Error::NotFoundError(NotFoundError(format!(
-                                    "No value found for {f} {:?}",
-                                    values
-                                ))))
-                            }
+                        if function.decl.subtype == FunctionSubtype::Relation {
+                            function.insert(values, Value::unit(), ts);
+                            Value::unit()
+                        } else if function.decl.subtype == FunctionSubtype::Constructor {
+                            let value = Value {
+                                #[cfg(debug_assertions)]
+                                tag: function.schema.output.name(),
+                                bits: self.unionfind.make_set(),
+                            };
+                            function.insert(values, value, ts);
+                            value
+                        } else {
+                            return Err(Error::NotFoundError(NotFoundError(format!(
```

Comment:
Nit: this should probably provide a different error message given this PR, since the only case this is possible is when there is a bug in our checker.

Reply:
I think it can still be triggered by merge functions reading a table, e.g.:

```
(function foo () i64)

(function bar () i64 :merge (foo))

(set (bar) 0)

(fail (set (bar) 1))
```

Reply:
Yeah- could add a TODO since this is a known issue
In the future it should be checked and disallowed